storage/raft_log_queue_test.go

```diff
@@ -0,0 +1,121 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+// Author: Bram Gruneir (bram+code@cockroachlabs.com)
+
+package storage
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/cockroachdb/cockroach/client"
+	"github.com/cockroachdb/cockroach/roachpb"
+	"github.com/cockroachdb/cockroach/util"
+	"github.com/cockroachdb/cockroach/util/leaktest"
+)
+
+// TestGetTruncatableIndexes verifies that the correctly returns when there are
+// indexes to be truncated.
+func TestGetTruncatableIndexes(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	store, _, stopper := createTestStore(t)
+	defer stopper.Stop()
+	if _, err := store.GetReplica(0); err == nil {
+		t.Error("expected GetRange to fail on missing range")
+	}
+
+	store.DisableRaftLogQueue(true)
+
+	// Test on a new range which should not have a raft group yet.
+	rngNew := createRange(store, 100, roachpb.RKey("a"), roachpb.RKey("c"))
+	if truncatableIndexes, oldestIndex, err := getTruncatableIndexes(rngNew); err != nil || truncatableIndexes != 0 ||
+		oldestIndex != 0 {
+		if err != nil {
+			t.Errorf("expected no error, got %s", err)
+		}
+		if truncatableIndexes != 0 {
+			t.Errorf("expected 0 for truncatable index, got %d", truncatableIndexes)
+		}
+		if oldestIndex != 0 {
+			t.Errorf("expected 0 for oldest index, got %d", oldestIndex)
+		}
+	}
+
+	r, err := store.GetReplica(1)
+	if err != nil {
+		t.Error(err)
+	}
+
+	firstIndex, err := r.FirstIndex()
+	if err != nil {
+		t.Error(err)
+	}
+
+	// Write a few keys to the range.
+	for i := 0; i < 10; i++ {
+		key := roachpb.Key(fmt.Sprintf("key%02d", i))
+		args := putArgs(key, []byte(fmt.Sprintf("value%02d", i)))
+		if _, err := client.SendWrapped(store.testSender(), nil, &args); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	if truncatableIndexes, oldestIndex, err := getTruncatableIndexes(r); err != nil || truncatableIndexes == 0 ||
+		oldestIndex <= firstIndex {
+		if err != nil {
+			t.Errorf("expected no error, got %s", err)
+		}
+		if truncatableIndexes == 0 {
+			t.Errorf("expected a value for truncatable index, got 0")
+		}
+		if oldestIndex <= firstIndex {
+			t.Errorf("expected oldest index (%d) to be greater than or equal to first index (%d)", oldestIndex,
+				firstIndex)
+		}
+	}
+
+	// Enable the raft log scanner and and force a truncation.
+	store.DisableRaftLogQueue(false)
+	store.ForceRaftLogScanAndProcess(t)
+
+	newFirstIndex, err := r.FirstIndex()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if newFirstIndex <= firstIndex {
+		t.Errorf("log was not correctly truncated, older first index:%d, current first index:%d", firstIndex,
+			newFirstIndex)
+	}
+
+	// This might be flaky, if so, we can remove this final part of the test, the functionality is tested in
+	// client_raft_log_queue as well.
+	util.SucceedsWithin(t, time.Second, func() error {
```

Comment:
time.Second is more than enough time in this case.  Flakiness in this case should be very rare, and I'd expect it to complete with the first pass.
time.Second is used all over still and this matches the succeedsWithin in client_raft_log_queue_test

I'm more than willing to the change the value, but I'm not sure which constant to use in this case.